CalPlayer/Dashes/PlayerDashManager.cs

```diff
@@ -13,6 +14,32 @@ public static bool FindByID(string id, out PlayerDashEffect dashEffect)
             return DashIdentificationTable.TryGetValue(id, out dashEffect);
         }
 
+        // A new function that allows to add in new modded dashes for Calamity to recognize, it attempts
+        // to make it as safe as possible.
+
+
+        /// <summary>
+        /// Attempts to add a dash effect into the dash manager. It should be executed in a Load() function.
+        /// </summary>
+        /// <param name="dashEffect">An instance of the desired dash effect to add.</param>
+
+        public static void TryAddDash(PlayerDashEffect dashEffect)
+        {
+            //If DashIdentificationTable is not loaded or modder tries to load an abstract type of
+            //PlayerDashEffect, then don't add the dash and stop the method.
+            if (DashIdentificationTable == null || dashEffect.GetType().IsAbstract) return;
+
+            string id = (string)dashEffect.GetType().GetProperty("ID").GetValue(null);
```

Comment:
Admittedly unrelated, but I loathe using reflection here to get IDs...
Maybe this is a good opportunity to move the system to a `ModType`? I'll investigate this personally.

Non-blocking.

Reply:
I think it can be a good idea, but I assume it'd be up to the Calamity devs as I only implemented it like that to ensure the class wouldn't break

Reply:
Yes, I would handle it.